### PR TITLE
Tweak: Apply filter for activate user in admin tool bar [ED-4309]

### PR DESCRIPTION
**Pull request by nisan250**

### CODE REVIEW

**Review by Nevoss — APPROVED** [review]

File: core/common/modules/connect/module.php [hunk 1]
```diff
@@ -210,10 +210,11 @@ protected function get_init_settings() {
 		if ( ! $connect ) {
 			return [];
 		}
-		return [
+
+		return apply_filters('elementor/connect/init_settings', [
```
Comment (matipojo) [inline core/common/modules/connect/module.php:214]: Remove from here, move it to filter only the top bar settings

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/modules/connect/module.php (hunk 1 above)
Comment (nisan250) [reply core/common/modules/connect/module.php:214]: Done

**Review by matipojo — APPROVED** [review]

File: core/common/modules/connect/module.php [hunk 2]
```diff
@@ -203,17 +203,4 @@ public function get_subscription_plans( $context ) {
 			],
 		];
 	}
-
-	protected function get_init_settings() {
```
Comment (matipojo) [inline core/common/modules/connect/module.php:207]: Restore, Why it's deleted?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: modules/admin-top-bar/module.php [hunk 3]
```diff
@@ -45,10 +46,22 @@ private function render_admin_top_bar() {
...
+		/** @var \Elementor\Core\Common\Modules\Connect\Apps\Library $library */
+		$library = Plugin::$instance->common->get_component( 'connect' )->get_app( 'library' );
+		if ( $library ) {
+			$settings = array_merge($settings, [
+				'is_user_connected' => $library->is_connected(),
+				'connect_url' => $library->get_admin_url( 'authorize' ),
+			]);
+		}
+
+		return apply_filters( 'elementor/admin-top-bar/init_settings', $settings );
```
Comment (matipojo) [inline modules/admin-top-bar/module.php:64]: Use do_action with set_settings.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: modules/admin-top-bar/module.php (hunk 3 above)
Comment (nisan250) [reply modules/admin-top-bar/module.php]: Done

File: core/common/modules/connect/module.php (hunk 2 above)
Comment (nisan250) [reply core/common/modules/connect/module.php:207]: Not sure why to restore it. it is a code that I added, but then we decided  to change the settings only for the Admin Top Bar, not widely

File: modules/admin-top-bar/module.php (hunk 3 above)
Comment (nisan250) [reply modules/admin-top-bar/module.php]: Done

File: core/common/modules/connect/module.php (hunk 2 above)
Comment (matipojo) [reply core/common/modules/connect/module.php:207]: The method `get_init_settings` existed before your changes.

File: modules/admin-top-bar/module.php [hunk 4]
```diff
@@ -45,11 +45,23 @@ private function render_admin_top_bar() {
 		</div>
 		<?php
 	}
-	protected function get_init_settings() {
-		return [
-			'is_administrator' => current_user_can( 'manage_options' ),
-		];
-	}
+
+	//protected function get_init_settings() {
```
Comment (matipojo) [inline modules/admin-top-bar/module.php:49]: Remove

File: modules/admin-top-bar/module.php [hunk 5]
```diff
@@ -81,11 +93,12 @@ private function enqueue_scripts() {
 	 *
 	 * Fired by `wp_dashboard_setup` action.
 	 *
-	 * @since 1.9.0
+	 * @since  1.9.0
```
Comment (matipojo) [inline modules/admin-top-bar/module.php:96]: ?

File: modules/admin-top-bar/module.php [hunk 6]
```diff
@@ -81,11 +93,12 @@ private function enqueue_scripts() {
...
 	 * @access public
 	 */
 	public function register_dashboard_widgets() {
-		wp_add_dashboard_widget( 'e-dashboard-widget-admin-top-bar', __( 'Elementor Top Bar', 'elementor' ), function () {} );
+		wp_add_dashboard_widget( 'e-dashboard-widget-admin-top-bar', __( 'Elementor Top Bar', 'elementor' ), function () {
```
Comment (matipojo) [inline modules/admin-top-bar/module.php:100]: ?

File: modules/admin-top-bar/module.php [hunk 7]
```diff
@@ -105,5 +118,23 @@ public function __construct() {
...
+				$settings = array_merge( $settings, [
+					'is_user_connected' => $library->is_connected(),
+					'connect_url' => $library->get_admin_url( 'authorize' ),
+				] );
+			}
+
+			$this->set_settings( $settings );
+
+			do_action( 'elementor/admin-top-bar/init', $this );
+		}, 12 /* After component 'connect' register the apps */, 1  );
```
Comment (matipojo) [inline modules/admin-top-bar/module.php:138]:
```suggestion
		}, 12 /* After component 'connect' register the apps */  );
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
Rename the PR to describe these changes.

File: modules/admin-top-bar/module.php (hunk 5 above)
Comment (nisan250) [reply modules/admin-top-bar/module.php:96]: Fixed

File: modules/admin-top-bar/module.php (hunk 6 above)
Comment (nisan250) [reply modules/admin-top-bar/module.php:100]: Fixed

File: modules/admin-top-bar/module.php (hunk 4 above)
Comment (nisan250) [reply modules/admin-top-bar/module.php:49]: Done.

File: core/common/modules/connect/module.php (hunk 2 above)
Comment (nisan250) [reply core/common/modules/connect/module.php:207]: The logic inside the get_init_settings is related to the admin top bar, after the changes in this PR and the PR in Pro it is not needed anymore.

File: modules/admin-top-bar/module.php (hunk 7 above)
Comment (nisan250) [reply modules/admin-top-bar/module.php:138]: Done

**Review by matipojo — APPROVED** [review]

**Pull request merged**